packages/jsii/lib/jsii-diagnostic.ts

```diff
@@ -816,3 +816,42 @@ export class JsiiDiagnostic implements ts.Diagnostic {
 export type DiagnosticMessageFormatter = (
   ...args: any[]
 ) => JsiiDiagnostic['messageText'];
+
+export function configureCategories(records: { [code: string]: string }) {
+  for (const [code, level] of Object.entries(records)) {
+    const diagCode = Code.lookup(diagnosticCode(code));
+    const category = diagnosticCategory(level);
+    if (!diagCode) {
+      throw new Error(`Unrecognized diagnostic code '${code}'`);
+    }
+    diagCode.category = category;
+  }
+}
+
+function diagnosticCategory(str: string): ts.DiagnosticCategory {
+  switch (str.trim().toLowerCase()) {
+    case 'error':
+      return ts.DiagnosticCategory.Error;
+    case 'warning':
+      return ts.DiagnosticCategory.Warning;
+    case 'suggestion':
+      return ts.DiagnosticCategory.Suggestion;
+    case 'message':
+      return ts.DiagnosticCategory.Message;
+    default:
+      throw new Error(`Unrecognized diagnostic category '${str}'`);
```

Comment:
Would be lovely if this can also refer to the key this unrecognized diagnostic was associated to (sometimes you fail to see the typo for wayyyyyyy too long).